### PR TITLE
feat: add create anonymous context to agent

### PR DESCRIPTION
**Pull request by denghongcai**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

`Application` and `Agent` class

##### Description of change

move `createAnonymousContext` method to `EggApplication` class

### CODE REVIEW

File: lib/egg.js [hunk 1]
```diff
@@ -456,6 +457,75 @@ class EggApplication extends EggCore {
...
   }
+
+  /**
+   * Create an anonymous context, the context isn't request level, so the request is mocked.
+   * then you can use context level API like `ctx.service`
+   * @member {String} EggApplication#createAnonymousContext
+   * @param {Request} req - if you want to mock request like querystring, you can pass an object to this function.
+   * @return {Context} context
+   */
+  createAnonymousContext(req) {
```
Comment (popomore) [inline lib/egg.js]: 放 agent 有点奇怪，不是 http 请求，只为了创建一个 ctx

Comment (denghongcai) [reply lib/egg.js]: 放 app 里其实也一样，也不是个 http 请求

**Comment by dead-horse** [comment]
有冲突

**Comment by denghongcai** [comment]
rebase 了

**Pull request merged**